Automattic/woocommerce-payments
# add default (= null) for classNames prop so it's optional for `WizardTaskItem` and `CollapsibleBody`

Fixes #9142

#### Changes proposed in this Pull Request
- Migrates the `WizardTaskItem` and `CollapsibleBody` components to typescript, explicitly defining the `className` as optional
- Remove redundant `className=null` prop in tax modal code – no longer needed.

Clients currently have to pass `classNames = { null }` to appease TypeScript. The prop is actually optional, only used if client wants to customise the `class` of generated markup.

See https://github.com/Automattic/woocommerce-payments/blob/develop/client/vat/form/tasks/vat-number-task.tsx for an example.

The goal of this PR is to streamline code like `VatNumberTask` so it doesn't need to pass meaningless `className={ null }`.

#### Testing instructions

* Run this branch and test any code that uses `WizardTaskItem` and `CollapsibleBody`, e.g. show the VAT modal. See the following PR for instructions for testing the tax documents / tax info modal:
  * https://github.com/Automattic/woocommerce-payments/pull/8974
- Hack any code using `WizardTaskItem` and `CollapsibleBody` and add a custom `className`, to confirm that the rendered component still supports arbitrary custom classes as before.
- Confirm that linters, TypeScript, code checks are all green ✅ and happy.

* ~~Hack on any code that uses `WizardTaskItem` and `CollapsibleBody`~~…
  * ~~Remove the `classNames={ null }`, ensure no errors/linter/typescript warnings and UI works as expected.~~
  * ~~Add custom classes via `classNames` and ensure they are rendered correctly in generated markup.~~

-------------------


**Post merge**
